src/Corax/Querying/Matches/SortingMatches/NumericalMaxHeapSorter.cs

```diff
@@ -0,0 +1,303 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using Sparrow.Server;
+using Voron.Util;
+#if DEBUG
+using Newtonsoft.Json;
+#endif
+
+namespace Corax.Querying.Matches.SortingMatches;
+
+internal unsafe ref struct NumericalMaxHeapSorter<TTermType, TSecondaryComparer> where TSecondaryComparer : IComparer<int>
```

Comment:
Add documentation here on assumptions of how and why this sorter can be used. If someone along the line has to decide into using a normal sorter and this sorter, they have to know why you would and you would not decide to use as a primitive by laying out the assumptions clearly.

Reply:
done